File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

int index;

bool cmp(string a, string b){
	
	if(a[index] == b[index])
        return a < b;

    return a[index] < b[index];
    
}

vector<string> solution(vector<string> strings, int n) {
	index = n;
    sort(strings.begin(), strings.end(), cmp);
    return strings;
}

File: 프로그래머스/lv1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
    
    return strings.sort((a,b) => {
        if(a[n] === b[n]){
            if(a < b)
                return -1;
            else
                return 1;
        }
        else{
            if(a[n] < b[n])
                return -1;
            else
                return 1;
        }
    })
}